test/s3-mock.ts

```diff
@@ -0,0 +1,64 @@
+import { Readable } from 'stream';
+import type { S3 } from '@aws-sdk/client-s3';
+import { afterEach } from '@jest/globals';
+
+interface S3Url {
+  Bucket: string;
+  Key: number;
+}
+
+let mockedObjects: any = {};
+let mockedTimestamps: any = {};
```

Comment:
please use `Record` and no `any` if possible. use `unknown` when no concrete type is possible.